src/Bicep.Core/Parser/Parser.cs

```diff
@@ -297,25 +303,112 @@ private SyntaxBase ParenthesizedExpression()
 
         private SyntaxBase FunctionCallOrVariableAccess()
         {
-            var identifier = this.Identifier(b => b.ExpectedFunctionName());
+            if (IsFunctionCall())
+            {
+                return GetFunctionCallSyntax();
+            }
+            // assumes variable call
+            return GetVariableAccessSyntax();
+        }
 
-            if (this.Check(TokenType.LeftParen) == false)
+        /// <summary>
+        /// Method that peeks current and next token, if current token is an identifier and the next is 
+        /// is left parenthesis then assumes a function call
+        /// </summary>
+        private bool IsFunctionCall()
+        {
+            if (reader.Peek().Type == TokenType.Identifier &&
+               reader.Position + 1 <= reader.Count - 1 &&
```

Comment:
>reader.Position + 1 <= reader.Count - 1 && [](start = 15, length = 42)

You should probably update the InvalidExpressions samples to add various incomplete instance function calls.